src/Features/Core/Portable/CodeFixes/FixAllOccurrences/FixAllProviderInfo.cs

```diff
@@ -47,7 +60,67 @@ public static FixAllProviderInfo Create(CodeFixProvider provider)
                 return null;
             }
 
-            return new FixAllProviderInfo(fixAllProvider, diagnosticIds, scopes);
+            return new CodeFixerFixAllProviderInfo(fixAllProvider, diagnosticIds, scopes);
+        }
+
+        /// <summary>
+        /// Gets an optional <see cref="FixAllProviderInfo"/> for the given suppression fix provider.
+        /// </summary>
+        private static FixAllProviderInfo CreateWithSuppressionFixer(ISuppressionFixProvider provider)
+        {
+            var fixAllProvider = provider.GetFixAllProvider();
+            if (fixAllProvider == null)
+            {
+                return null;
+            }
+
+            var scopes = fixAllProvider.GetSupportedFixAllScopes();
+            if (scopes == null || scopes.IsEmpty())
+            {
+                return null;
+            }
+
+            return new SuppressionFixerFixAllProviderInfo(fixAllProvider, provider, scopes);
+        }
+
+        public abstract bool CanBeFixed(Diagnostic diagnostic);
+
+        private class CodeFixerFixAllProviderInfo : FixAllProviderInfo
+        {
+            private readonly IEnumerable<string> _supportedDiagnosticIds;
+
+            public CodeFixerFixAllProviderInfo(
+                FixAllProvider fixAllProvider,
+                IEnumerable<string> supportedDiagnosticIds,
+                IEnumerable<FixAllScope> supportedScopes)
+                : base (fixAllProvider, supportedScopes)
+            {
+                this._supportedDiagnosticIds = supportedDiagnosticIds;
+            }
+
+            public override bool CanBeFixed(Diagnostic diagnostic)
+            {
+                return _supportedDiagnosticIds.Contains(diagnostic.Id);
+            }
+        }
+
+        private class SuppressionFixerFixAllProviderInfo : FixAllProviderInfo
+        {
+            private readonly Func<Diagnostic, bool> _canBeSuppressedOrTriaged;
+
+            public SuppressionFixerFixAllProviderInfo(
+                FixAllProvider fixAllProvider,
+                ISuppressionFixProvider suppressionFixer,
+                IEnumerable<FixAllScope> supportedScopes)
+                : base(fixAllProvider, supportedScopes)
+            {
+                this._canBeSuppressedOrTriaged = suppressionFixer.CanBeSuppressed;
+            }
+
+            public override bool CanBeFixed(Diagnostic diagnostic)
+            {
+                return _canBeSuppressedOrTriaged(diagnostic);
```

Comment:
Do we really need a lambda here? Can't we just save suppressionFixer in a field and call suppressionFixer.CanBeSuppressed(diag)?